aim/api/status.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (c) 2016 Cisco Systems
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+
+class Status(object):
+
+    def is_build(self):
+        """Returns True if the current status is transient."""
+        raise
+
+    def is_error(self):
+        """Returns True if current status is an error state."""
+        raise
+
+
+class AciStatus(Status):
+    """Status of an AIM resource that is mapped to ACI object.
+
+    Following attributes are available:
+    * create_status - Indicates whether ACI object was created or not
```

Comment:
I was discussing this with Sumit earlier today, and it seemed that setting the status of a resource to "BUILD" when changes are made to it was probably not useful. Also this situation is likely to be very transient. But I guess if that sync fails for some reason (e.g. update violates some check in ACI), then it would make sense to report that. I'll update the diffs.